common/tools/dev-tool/src/util/customization/classes.ts

```diff
@@ -0,0 +1,357 @@
+    }
+
+    const originalProperty = originalClassDeclaration.getProperty(propertyName);
+    const propertyComments = getComments(customProperty, originalProperty);
+
+    // If the property already exists in the original declaration, we'll replace it
+    originalProperty?.remove();
+    addPropertyToClass(customProperty, originalClassDeclaration, propertyComments);
+    // originalClassDeclaration.addProperty(customProperty.getStructure());
+  }
+
+  // Get custom methods
+  const customMethods = customClassDeclaration.getMethods();
+  for (const customMethod of customMethods) {
+    const methodName = customMethod.getName();
+
+    // If the method already exists in the original declaration, we'll replace it
+    const originalMethod = originalClassDeclaration.getMethod(methodName);
+    augmentMethod(originalMethod, customMethod, originalClassDeclaration);
+  }
+
+  // Get custom constructors
+  const customConstructors = customClassDeclaration.getConstructors();
+  for (const customConstructor of customConstructors) {
+    const originalConstructor = originalClassDeclaration
+      .getConstructors()
+      .find((c) => !isOverload(c.getStructure()));
+    augmentConstructor(customConstructor, originalConstructor, originalClassDeclaration);
+  }
+}
+
+export function augmentConstructor(
+  customConstructor: ConstructorDeclaration,
+  originalConstructor: ConstructorDeclaration | undefined,
+  originalClassDeclaration: ClassDeclaration
+) {
+  const constructorComments = getComments(customConstructor, originalConstructor);
+  addConstructorsToClass(customConstructor, originalClassDeclaration, constructorComments);
+}
+
+export function augmentMethod(
+  originalMethod: MethodDeclaration | undefined,
+  customMethod: MethodDeclaration,
+  originalClass: ClassDeclaration
+) {
+  const methodComments = getComments(customMethod, originalMethod);
+  // custom is adding a new method this is a new method on the class, we'll add it to original
+  if (!originalMethod) {
+    addMethodToClass(customMethod, originalClass, methodComments);
+    return;
+  }
+
+  // Original method with the same name exists so
+  // we need to check if the custom method is using the original method
+  // to determine if we need to augment or replace
+  if (isAugmentingMethod(customMethod)) {
+    convertToPrivateMethod(originalMethod, originalClass);
+    addMethodToClass(customMethod, originalClass, methodComments);
+  } else {
+    // This is not using the original method so we'll replace it
+    originalMethod.remove();
+    addMethodToClass(customMethod, originalClass, methodComments);
+  }
+}
+
+export function isAugmentingConstructor(
+  customConstructor: ConstructorDeclaration,
+  originalConstructor?: ConstructorDeclaration
+): boolean {
+  const customConstructorContent = customConstructor.getBody()?.getFullText();
+
+  if (!originalConstructor) {
+    // If there is no original constructor, we'll just copy the custom one
+    return false;
+  }
```

Comment:
nitpick: exit earlier?
```suggestion
if (!originalConstructor) {
    // If there is no original constructor, we'll just copy the custom one
    return false;
  }
  const customConstructorContent = customConstructor.getBody()?.getFullText();
```